spec/preprocessor_spec.rb

```diff
@@ -17,7 +17,7 @@
   describe "inset text" do
     context "when there is an inline piece of inset text" do
       let(:input) do
-        <<~MD
+        <<~MD.freeze
```

Comment:
nice. I'm thinking I might create separate specs with frozen strings to highlight what we're doing here, rather than add it into existing specs - otherwise I'm going to forget why we did this...